config/crd/bases/infrastructure.cluster.x-k8s.io_awsmachinepools.yaml

```diff
@@ -0,0 +1,307 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.2.9
+  creationTimestamp: null
+  name: awsmachinepools.infrastructure.cluster.x-k8s.io
+spec:
+  group: infrastructure.cluster.x-k8s.io
```

Comment:
I think we want this to be under `exp.infrastructure.x-k8s.io` if we're aiming to be consistent https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/exp/PROJECT

Reply:
there was a bit of discussion here: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1724#discussion_r461696188

I think we're in favor of keeping the api group the same just because migrating between two groups is significantly harder than migrating between two versions. so while most of the code itself will be `exp`, the API group is not.

Reply:
That said, there is work to do here to put the controller behind the experimental flag. cc @mytunguyen @NicoleYson

Reply:
Ok, I am thoroughly confused by all this but it's my first endeavor into experimental features so I'll assume it makes sense 😄

Reply:
:D let me try explaining it slightly differently:

this is definitely an experimental feature that should be behind the MachinePool feature gate, 100%. eventually, we'd like to move it out of experimental at some point in the future, which will likely mean removing the feature gate and shuffling some code around. that's totally fine and straightforward. 

however, if the api group needs to change on the CRD (from exp.infrastructure... to infrastructure...), that effectively represents a brand new CRD from K8S' perspective, so things like conversion webhooks, won't be usable to us. upgrading an existing cluster from the exp.infrastructure machinepool to the infrastructure machinepool will likely be disruptive, and that is avoidable if we use the same API group.

Reply:
So is it a mistake that on [CAPI](https://github.com/kubernetes-sigs/cluster-api/blob/60ce90a7bf173d008b39492be45e1a1fef02bcdd/config/crd/bases/exp.cluster.x-k8s.io_machinepools.yaml#L11) and [CAPZ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/config/crd/bases/exp.infrastructure.cluster.x-k8s.io_azuremachinepools.yaml#L11) we made the group `exp.infrastructure.cluster.x-k8s.io`?

Reply:
I think dropping that `exp.` on all counts would remove some future headaches - so yes, I think so.

Reply:
We shouldn't use `exp` in API groups, an experiment is only defined as such if it lives under exp/ folder and it has a feature gate associated with it.